app/models/user.rb

```diff
@@ -26,7 +26,7 @@ def cart
   
   has_one :owned_company, foreign_key: "user_id", class_name: "Company" 
   
-  has_one :company_user
+  has_one :company_user, dependent: :destroy
```

Comment:
esse dependent: :destroy 
não irá destruir a Company que o usuario for empregado, caso ele seja deletado ?

Reply:
acho que so vai delete a company_user que é ligado ao user. nao a company

Reply:
se for deletado o owner, a company dele é automaticamente destruida então. 
isso?